src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -858,12 +859,13 @@ private void GenerateProperties(Type type, JsonSchema schema, JsonSchemaResolver
 #if !LEGACY
             var members = type.GetTypeInfo()
                 .DeclaredFields
-                .Where(f => !f.IsPrivate && !f.IsStatic)
+                .Where(f => !f.IsPrivate && !f.IsStatic || f.IsDefined(typeof(DataMemberAttribute)))
                 .OfType<MemberInfo>()
                 .Concat(
                     type.GetTypeInfo().DeclaredProperties
                     .Where(p => (p.GetMethod?.IsPrivate != true && p.GetMethod?.IsStatic == false) ||
-                                (p.SetMethod?.IsPrivate != true && p.SetMethod?.IsStatic == false))
+                                (p.SetMethod?.IsPrivate != true && p.SetMethod?.IsStatic == false) ||
+                                p.IsDefined(typeof(DataMemberAttribute)))
```

Comment:
Can you also implement this for the "legacy" part?

Reply:
Nonlegacy is OK. But after I change the NJsonSchema project target to net40, I can't debug the project for the "legacy" part. Even if I create a new net40 console project for test, there is a weird [CS0103] error.

Reply:
Ok then it's fine.